File: remove-duplicates-from-sorted-array.py
# 1 1 2 3 3 4 5 5 5 9 --
# 1 2 3 4 5 9
#           l 
#                   r
# 1
# l
#   r
# if al = ar: move l, then a[l] = a[r]
# at the end return l + 1
# 1 2 2
# 1 2
#   l
#     r
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        l = 0
        for r in range(1, len(nums)):
            if nums[l] != nums[r]:
                l += 1
                nums[l] = nums[r]
        return 1 + l

File: lc/remove-duplicates-from-sorted-array.go
/*
0,1,1,1,1,2,2,3,3,4
a   
          b
compare b to a: if different, put nums[b] into nums[a+1]; a++
ans: a+2

corner: if len is 1 just return 0
0
a
  b
return 

5 5-8 8
  a
      b
*/
func removeDuplicates(nums []int) int {
    n := len(nums)
    a, b := 0, 1   
    for ; b < n; b ++ {
        if nums[b] != nums[a] {
            a++
            nums[a] = nums[b]
        }
    }
    return a + 1
}